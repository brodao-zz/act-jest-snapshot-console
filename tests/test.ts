const serializer = require('../src/index');

expect.addSnapshotSerializer(serializer);

const { wrap } = serializer;

const myLog = (msg: string) => {
	console.log('Executando myFunctionTest');
	console.log(`Msg = ${msg}`);
	console.log('% % % %');
};

const myLogWithLevel = (level: 'error' | 'info' | 'warn' | 'log') => {
	console[level](`Mensagem de nÃ­vel '${level}'`);
};

test('before', () => {
	expect(myLog('before')).toMatchSnapshot();
});

test('after', () => {
	expect(wrap(() => myLog('after'))).toMatchSnapshot();
});

test('error', () => {
	expect(wrap(() => myLogWithLevel('error'))).toMatchSnapshot();
});

test('info', () => {
	expect(wrap(() => myLogWithLevel('info'))).toMatchSnapshot();
});

test('log', () => {
	expect(wrap(() => myLogWithLevel('log'))).toMatchSnapshot();
});

test('warn', () => {
	expect(wrap(() => myLogWithLevel('warn'))).toMatchSnapshot();
});
