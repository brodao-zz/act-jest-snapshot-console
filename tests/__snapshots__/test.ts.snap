// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`after 1`] = `
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
L:Executando myFunctionTest
L:Msg = after
L:% % % %

`;

exports[`before 1`] = `undefined`;

exports[`error 1`] = `
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
E:Mensagem de nível 'error'

`;

exports[`info 1`] = `
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
I:Mensagem de nível 'info'

`;

exports[`log 1`] = `
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
L:Mensagem de nível 'log'

`;

exports[`log 2`] = `
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
W:Mensagem de nível 'warn'

`;
